src/Lecture2.hs

```diff
@@ -70,7 +77,13 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt 0 (x:xs) = (Just x, xs)
+removeAt n (x:xs) = (a, x: b)
+  where
+    (a, b) = removeAt (n - 1) xs
```

Comment:
It looks like this function hangs when given a negative number and infinite list at the same time 🤔